python/activator/exception.py

```diff
@@ -107,3 +109,28 @@ class IgnorableVisit(ValueError):
     activator.visit.SummitVisit
     activator.visit.FannedOutVisit
     """
+
+
+class InvalidPipelineError(ValueError):
+    """Exception raised if a pipeline cannot be loaded or configured.
+    """
+
+
+class NoGoodPipelinesError(RuntimeError):
```

Comment:
I don't see anything catching this specific exception. The `MWI._run_preprocessing` method is documented to raise this exception, and it is called from `MWI.prep_butler`, but the latter does not seem to handle `NoGoodPipelinesError` and does not document that it can raise it. Do the clients of `prep_butler` need to handle generic `RuntimeError` or `Exception`?

Reply:
You are correct that `NoGoodPipelinesError` is not caught -- except where chained to another exception, it will propagate up to the top-level (Flask) exception handler. While it's an unlikely scenario in preprocessing (`prep_butler` itself controls the inputs for the pipelines we run there), I suppose we would want to follow the same policy of "try a simple main pipeline anyway".